### PR TITLE
Fix new fissures not being sorted

### LINKED ISSUE #140: Fissures added since page load are not sorted
### Fissures added since page load are not sorted
---

**Summary (short):**
On page load, fissures are sorted by era in ascending order. However, new fissures are appended to the bottom of the section instead of adhering to the order. Refreshing the page is the current workaround.

---
**Description:**
`updateFissure()` has the following sort function:
```javascript
    fissures.sort((a, b) => {
      const tierA = a.tierNum;
      const tierB = b.tierNum;
      if (tierA < tierB) { return -1; }
      if (tierA > tierB) { return 1; }
      return 0;
    });
```
Adhere to this order.  New fissures should go to the bottom of the relevant era.

---
**If issue, Reproduction:**
Wait until a new non-Axi fissure appears while there is an ongoing Axi fissure.

---
**Mockups, screenshots, evidence:**

---

(Check one)
- [x] Issue
- [ ] Suggestion

### PR DESCRIPTION
**Pull request by dsinn**
Closes #140.

### Fix new fissures not being sorted
---
**Summary (short):**
See issue.

---
**Description:**
See issue.

---
**Fixes issue (include link):**
#140

---
**Mockups, screenshots, evidence:**

---

(Check one)
- [x] Issue fix
- [ ] Suggestion fulfillment

### CODE REVIEW

File: assets/js/updaters.js [hunk 1]
```diff
@@ -753,7 +745,20 @@ function updateFissure() {
           }
         });
         if (!filtered) {
-          $('#fissurebody').before(fissureRow);
+          let added = false;
+
+          // Order by tier number
+          $('#fissureList .fissure').each(function () {
```
[the remark this answers is not in the document]
Comment (TobiTenno) [reply assets/js/updaters.js]: just need to use arrow functions, and it's fine

[the remark this answers is not in the document]
Comment (TobiTenno) [reply assets/js/updaters.js]: just need to use arrow functions, and it's fine

File: assets/js/updaters.js [hunk 2]
```diff
@@ -753,7 +745,21 @@ function updateFissure() {
           }
         });
         if (!filtered) {
-          $('#fissurebody').before(fissureRow);
+          let added = false;
+
+          // Order by tier number
+          $('#fissureList .fissure').each((index, existingFissure) => {
+            if ($(existingFissure).attr('data-tiernum') > fissure.tierNum) {
```
Comment (TobiTenno) [inline assets/js/updaters.js:752]: Could you grab it's index in the list of existing fissures, then using that, insert it before the next I'd if it exists, and if not, then add it at the end?

**Review by TobiTenno — CHANGES_REQUESTED** [review]

File: assets/js/updaters.js (hunk 2 above)
Comment (dsinn) [reply assets/js/updaters.js:752]: I'm not sure what you mean by that, sorry. What indices and values would we be storing here? Could you provide some pseudocode?

Comment (TobiTenno) [reply assets/js/updaters.js:752]: Not really, I'm on a phone. We'd be using the IDs of the fissure rows, so finding what should be before or after it and inserting as applicable

Comment (dsinn) [reply assets/js/updaters.js:752]: Still don't know what you mean.  Please discuss with me on Discord when you have time, I guess.

Comment (TobiTenno) [reply assets/js/updaters.js:752]: This is fine for the current solution. I think it's actually just as efficient as what i was envisioning.

**Review by TobiTenno — APPROVED** [review]

**Pull request merged**